### PR TITLE
Only run pytest if test files are found in the test directories

### LINKED ISSUE #4979: Python modules: Remove the need for adding dummy tests for CI
## Tell us about the problem you're trying to solve
Our gradle python plugin [automatically adds a task for running pytest](https://github.com/airbytehq/airbyte/blob/master/buildSrc/src/main/groovy/airbyte-python.gradle#L88) if any python module has a `unit_tests` or `integration_tests` directory. However, if there are no tests in that dir, pytest [exits with exit code `5`](https://github.com/pytest-dev/pytest/issues/812) which gradle takes to mean that the process failed, since it's non-zero. 

This means that if a module doesn't need a unit or integration test, it still needs to add a dummy test file containing something like:

```
def make_ci_pass_test():
  assert True
```

just to make pytest exit with a zero exit code. 

## Describe the solution you’d like
Edit the Gradle plugin to only add the unit test tasks if there are files which start with `test_*.py` or `*_test.py` according to Pytest's test discovery rules.

### PR DESCRIPTION
**Pull request by sherifnada**
## What
Closes #4979 

## How
Only add test tasks if files matching pytest [test naming rules](https://docs.pytest.org/en/6.2.x/goodpractices.html#conventions-for-python-test-discovery) are found in the unit_tests or integration_tests directories

### CODE REVIEW

File: airbyte-integrations/connectors/source-mongodb/build.gradle [hunk 1]
```diff
@@ -36,3 +36,4 @@ airbyteStandardSourceTestFile {
 dependencies {
     implementation files(project(':airbyte-integrations:bases:base-standard-source-test-file').airbyteDocker.outputs)
 }
+
```
Comment (davinchia) [inline airbyte-integrations/connectors/source-mongodb/build.gradle:39]: should the empty test2 file be deleted?

File: buildSrc/src/main/groovy/airbyte-python.gradle [hunk 2]
```diff
@@ -8,6 +10,46 @@ class AirbytePythonConfiguration {
     String moduleDirectory
 }
 
+class Helpers {
```
Comment (davinchia) [inline buildSrc/src/main/groovy/airbyte-python.gradle:13]: bold groovy scripting

File: airbyte-integrations/connectors/source-mongodb/build.gradle (hunk 1 above)
Comment (sherifnada) [reply airbyte-integrations/connectors/source-mongodb/build.gradle:39]: oh woops, remnant of an experiment

File: buildSrc/src/main/groovy/airbyte-python.gradle [hunk 3]
```diff
@@ -8,6 +10,46 @@ class AirbytePythonConfiguration {
...
+        So we use this method to leverage pytest's test discovery rules (https://docs.pytest.org/en/6.2.x/goodpractices.html#conventions-for-python-test-discovery)
+        to selectively run pytest based on whether there seem to be test files in that directory. 
+       
+       
+        See https://github.com/airbytehq/airbyte/issues/4979 for original context
+        """
+        if (project.file(testFilesDirectory).exists()) {
+
+            project.projectDir.toPath().resolve(testFilesDirectory).traverse(type: FileType.FILES, nameFilter: ~/(^test_.*|.*_test)\.py$/) { file ->
+                project.task(taskName, type: PythonTask, dependsOn: taskDependencies) {
```
Comment (davinchia) [inline buildSrc/src/main/groovy/airbyte-python.gradle:35]: what happens here if there are multiple files found? will this add a task for each file?

Comment (sherifnada) [reply buildSrc/src/main/groovy/airbyte-python.gradle:35]: the `return TERMINATE` ensures that traversal stops if a file is found, and so this is called at most once. Added as a comment.

Comment (davinchia) [reply buildSrc/src/main/groovy/airbyte-python.gradle:35]: thanks!

**Review by davinchia — APPROVED** [review]
🍇 🚀 

grapes was the closest emoji to groovy I could find.

**Review by keu — APPROVED** [review]

**Pull request merged**